packages/cozy-stack-client/src/DocumentCollection.js

```diff
@@ -178,19 +180,39 @@ export default class DocumentCollection {
   }
 
   async toMangoOptions(selector, options = {}) {
-    const indexFields = this.getIndexFields({ ...options, selector })
-    const indexId = options.indexId || (await this.getIndexId(indexFields))
+    let { sort } = options
     const { fields, skip = 0, limit } = options
-    // Mango wants an array of single-property-objects..
-    const sortOrders = options.sort
-      ? uniq(Object.values(options.sort))
-      : ['asc']
-    if (sortOrders.length > 1)
-      throw new Error('Mango sort can only use a single order (asc or desc).')
-    const sortOrder = sortOrders[0]
-    const sort = options.sort
-      ? indexFields.map(f => ({ [f]: sortOrder }))
-      : undefined
+
+    if (sort && !Array.isArray(sort)) {
+      console.warn(
+        'Passing an object to the "sort" is deprecated, please use an array instead.'
+      )
+      sort = transform(
+        sort,
+        (acc, order, field) => {
+          acc.push({ [field]: order })
+          return acc
+        },
+        []
+      )
```

Comment:
```js
sort = Object.entries(sort).map(([field, order]) => ({ [field]: order }))
```
?

Reply:
Yeah I don't mind either way. Not knowing how a lodash function works is more or less true of all lodash functions though :) But happy to change it if you think the native version makes the code clearer!